packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1440,89 +1478,206 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
+
+      void _closed(void value) {
+        setState(() {
+          _currentBottomSheet = null;
+        });
+
+        if (!isScrollControlled && animationController.status !=AnimationStatus.dismissed) {
+          _dismissedBottomSheets.add(bottomSheet);
+        } else if (isScrollControlled) {
+          if (!bottomSheet.isPersistent && _bottomSheetScrollController.animationStatus != AnimationStatus.completed) {
+            _dismissedBottomSheets.add(bottomSheet);
+          }
+          _bottomSheetScrollController.dispose();
+        }
+        completer.complete();
+      }
+
+      if (closing != null)
+        closing.then(_closed);
+      else
+        _closed(null);
     }
 
-    final LocalHistoryEntry entry = isLocalHistoryEntry
-      ? LocalHistoryEntry(onRemove: _removeCurrentBottomSheet)
-      : null;
+    final LocalHistoryEntry entry = isPersistent
+      ? null
+      : LocalHistoryEntry(onRemove: _removeCurrentBottomSheet);
 
-    bottomSheet = _PersistentBottomSheet(
+    bottomSheet = _StandardBottomSheet(
       key: bottomSheetKey,
-      animationController: controller,
-      enableDrag: isLocalHistoryEntry,
+      animationController: animationController,
+      enableDrag: isScrollControlled == false && !isPersistent,
       onClosing: () {
         assert(_currentBottomSheet._widget == bottomSheet);
-        if (isLocalHistoryEntry)
-          entry.remove();
-        else
+        if (isPersistent) {
           _removeCurrentBottomSheet();
+        } else if (!removedEntry) {
+          assert(entry != null);
+          removedEntry = true;
+          entry.remove();
+        }
       },
       onDismissed: () {
         if (_dismissedBottomSheets.contains(bottomSheet)) {
-          bottomSheet.animationController.dispose();
           setState(() {
             _dismissedBottomSheets.remove(bottomSheet);
           });
         }
       },
       builder: builder,
+      scrollController: _bottomSheetScrollController,
+      isPersistent: isPersistent,
     );
 
-    if (isLocalHistoryEntry)
+    if (!isPersistent)
       ModalRoute.of(context).addLocalHistoryEntry(entry);
 
     return PersistentBottomSheetController<T>._(
       bottomSheet,
       completer,
-      isLocalHistoryEntry ? entry.remove : _removeCurrentBottomSheet,
+      entry != null
+        ? entry.remove
+        : _removeCurrentBottomSheet,
       (VoidCallback fn) { bottomSheetKey.currentState?.setState(fn); },
-      isLocalHistoryEntry,
+      !isPersistent,
     );
   }
 
-  /// Shows a persistent material design bottom sheet in the nearest [Scaffold].
+  /// Shows a material design bottom sheet in the nearest [Scaffold]. To show
+  /// a persistent bottom sheet, use the [Scaffold.bottomSheet].
   ///
   /// Returns a controller that can be used to close and otherwise manipulate the
   /// bottom sheet.
   ///
+  /// The `isScrollControlled` parameter specifies whether this is a route for
```

Comment:
What does "this is a route" refer to here?

Reply:
Copy paste error.